compiler/code-gen/vertex-compiler.cpp

```diff
@@ -1454,14 +1453,60 @@ void compile_function_resumable(VertexAdaptor<op_function> func_root, CodeGenera
 
 }
 
+void compile_start_fork_function_in_light_mode(VertexAdaptor<op_function> func_root, CodeGenerator &W) {
+
+  FunctionPtr func = func_root->func_id;
+
+  const auto var_name_gen = [](CodeGenerator &W, VarPtr var) { W << VarName(var); };
+  //CALL FUNCTION
+  W << FunctionDeclaration(func, false) << " " << BEGIN;
+  if (func->has_global_vars_inside) {
+    W << PhpMutableGlobalsAssignCurrent() << NL;
+  }
+  W << "vk::final_action action([]{KphpForkContext::get().scheduler.mark_current_fork_as_ready();});" << NL;
```

Comment:
Let's recheck that it's working in case when current function is called from other fork. It's a bit strange that we change codegen of function we want to fork. It should be redundant when we use C++-20 coroutines.

Reply:
I think it's better to determine that fork is ready outside of its logic, i.e. somewhere in scheduler